src/main/java/Parser.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Represents a parser that helps to make sense of the user commands.
+ */
+public class Parser {
+    /**
+     * Reads the user input and returns the corresponding command according to the command
+     * given by the user.
+     *
+     * @param input The user input into the program.
+     * @return The corresponding command for the user input.
+     * @throws DukeException
+     */
+    public static Command parse(String input) throws DukeException {
+        input = input.toLowerCase();
+        String[] inputWords = input.split(" ");
+
+        if (inputWords.length == 0) {
+            throw new DukeException("Sorry, I don't understand what you are saying! D=\n"
+                    + "Type \"help\" to view the list of commands you can use!");
+        }
+
+        String commandWord = inputWords[0];
+
+        if (commandWord.equals("bye") && inputWords.length == 1) {
+            return new ExitCommand();
+        }
+
+        if (commandWord.equals("help") && inputWords.length == 1) {
+            return new HelpCommand();
+        }
+
+        if (commandWord.equals("list") && inputWords.length == 1) {
+            return new ListCommand();
+        }
+
+        if (commandWord.equals("find")) {
+            if (inputWords.length == 1) {
+                throw new DukeException("You need to specify what keyword to search! \n" +
+                        "eg find book");
+            }
+
+            String keyword = input.substring(4);
+            return new FindCommand(keyword);
+        }
+
+
+        if (commandWord.equals("done")) {
+            if (inputWords.length != 2) {
+                throw new DukeException("You need to specify which task to mark done! \n"
+                        + "eg done 1");
+            }
+
+            int i = input.charAt(5) - 48;
+            return new DoneCommand(i);
+        }
+
+        if (commandWord.equals("delete")) {
+            if (inputWords.length != 2) {
+                throw new DukeException("You need to specify which task to delete!\n"
+                        + "eg delete 1");
+            }
+
+            int i = input.charAt(7) - 48;
+            return new DeleteCommand(i);
+        }
+
+        if (commandWord.equals("todo")) {
+            if (inputWords.length == 1) {
+                throw new DukeException("You need to specify your todo task!\n"
+                        + "eg todo read book");
+            }
```

Comment:
For the separation, you could put inplace dividers as comments to 'prettier' your code. Maybe like a form of comments with format. I have found it helpful when I pick up anothers' person script to work on when they have extremely beautiful comments to make the code readable. I find that extremely easy to read as compared to just if and elses.